CMakeLists.txt

```diff
@@ -0,0 +1,188 @@
+                sol/table.hpp
+                sol/table_iterator.hpp
+                sol/thread.hpp
+                sol/tie.hpp
+                sol/traits.hpp
+                sol/tuple.hpp
+                sol/types.hpp
+                sol/unsafe_function.hpp
+                sol/userdata.hpp
+                sol/usertype_core.hpp
+                sol/usertype.hpp
+                sol/usertype_metatable.hpp
+                sol/usertype_traits.hpp
+                sol/variadic_args.hpp
+                sol/variadic_results.hpp
+                sol/wrapper.hpp)
+
+if (NOT MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning -Wno-unknown-warning-option -Wall -Wextra -Wpedantic -pedantic -pedantic-errors -Wno-noexcept-type -ftemplate-depth=1024")
+endif()
+
+set(EXAMPLES_SRC examples/any_return.cpp
+                 examples/basic.cpp
+                 examples/calling_lua_functions.cpp
+                 examples/config.cpp
+                 examples/containers_as_table.cpp
+                 examples/containers.cpp
+                 examples/container_usertype_as_container.cpp
+                 examples/coroutine.cpp
+                 examples/customization.cpp
+                 examples/dynamic_object.cpp
+                 examples/environments.cpp
+                 examples/environment_snooping.cpp
+                 examples/environments_on_functions.cpp
+                 examples/functions.cpp
+                 examples/multi_results.cpp
+                 examples/namespacing.cpp
+                 examples/optional_with_iteration.cpp
+                 examples/overloading.cpp
+                 examples/overloading_with_fallback.cpp
+                 examples/protected_functions.cpp
+                 examples/require.cpp
+                 examples/runtime_additions.cpp
+                 examples/script_error_handling.cpp
+                 examples/self_call.cpp
+                 examples/self_from_lua.cpp
+                 examples/stack_aligned_function.cpp
+                 examples/static_variables.cpp
+                 examples/tables.cpp
+                 examples/usertype_advanced.cpp
+                 examples/usertype_automatic_operators.cpp
+                 examples/usertype_bitfields.cpp
+                 examples/usertype_call_from_c++.cpp
+                 examples/usertype.cpp
+                 examples/usertype_dynamic_getter_setter.cpp
+                 examples/usertype_initializers.cpp
+                 examples/usertype_simple.cpp
+                 examples/usertype_special_functions.cpp
+                 examples/usertype_var.cpp
+                 examples/variables.cpp
+                 examples/variadic_args.cpp)
+source_group(EXAMPLES FILES ${EXAMPLES_SRC})
+foreach(example_source_file ${EXAMPLES_SRC})
+message(STATUS "${example_source_file}")
+  get_filename_component(example_name ${example_source_file} NAME_WE)
+  add_executable(${example_name} ${example_source_file} ${HEADER_SRCS})
```

Comment:
can we have it so the examples are called like "example-variadic_args", "example-usertype" etc? Otherwise it gets difficult to find the test executable in the alphabetically ordered build target drop down lists of the ide's.